### ISSUE TITLE
pip install TA-Lib Error ->   Failed building wheel for TA-Lib

### ISSUE THREAD

**Issue by focamonca** [freqtrade/freqtrade#4245]
## Describe your environment

  * Operating system: Rapberry pi3 
  * Python Version: 3.7.3
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:
when I try to install TA-lib I get tons of errors
pip install TA-Lib Error ->   Failed building wheel for TA-Lib
### Steps to reproduce:

  1. pip install TA-Lib Error

  
### Observed Results:

Failed building wheel for TA-Lib
written in red and then loads of errors

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
pi@raspberrypi:~ $ pip3 install TA-Lib
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting TA-Lib
  Using cached https://files.pythonhosted.org/packages/ac/cf/681911aa31e04ba171ab4d523a412f4a746e30d3eacb1738799d181e028b/TA-Lib-0.4.19.tar.gz
Requirement already satisfied: numpy in /usr/lib/python3/dist-packages (from TA-Lib) (1.16.2)
Building wheels for collected packages: TA-Lib
  Running setup.py bdist_wheel for TA-Lib ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-fugao5k4/TA-Lib/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-gdrcag0s --python-tag cp37:
  /tmp/pip-install-fugao5k4/TA-Lib/setup.py:71: UserWarning: Cannot find ta-lib library, installation may fail.
    warnings.warn('Cannot find ta-lib library, installation may fail.')
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-armv7l-3.7
  creating build/lib.linux-armv7l-3.7/talib
  copying talib/abstract.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/test_func.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/stream.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/test_abstract.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/test_data.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/deprecated.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/test_stream.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/test_pandas.py -> build/lib.linux-armv7l-3.7/talib
  copying talib/__init__.py -> build/lib.linux-armv7l-3.7/talib
  running build_ext
  building 'talib._ta_lib' extension
  creating build/temp.linux-armv7l-3.7
  creating build/temp.linux-armv7l-3.7/talib
  arm-linux-gnueabihf-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include -I/usr/local/include -I/opt/include -I/opt/local/include -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.7m -c talib/_ta_lib.c -o build/temp.linux-armv7l-3.7/talib/_ta_lib.o
  talib/_ta_lib.c:611:10: fatal error: ta-lib/ta_defs.h: File o directory non esistente
   #include "ta-lib/ta_defs.h"
            ^~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for TA-Lib
  Running setup.py clean for TA-Lib
Failed to build TA-Lib
Installing collected packages: TA-Lib
  Running setup.py install for TA-Lib ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-fugao5k4/TA-Lib/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-_opvumy8/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    /tmp/pip-install-fugao5k4/TA-Lib/setup.py:71: UserWarning: Cannot find ta-lib library, installation may fail.
      warnings.warn('Cannot find ta-lib library, installation may fail.')
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-armv7l-3.7
    creating build/lib.linux-armv7l-3.7/talib
    copying talib/abstract.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/test_func.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/stream.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/test_abstract.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/test_data.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/deprecated.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/test_stream.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/test_pandas.py -> build/lib.linux-armv7l-3.7/talib
    copying talib/__init__.py -> build/lib.linux-armv7l-3.7/talib
    running build_ext
    building 'talib._ta_lib' extension
    creating build/temp.linux-armv7l-3.7
    creating build/temp.linux-armv7l-3.7/talib
    arm-linux-gnueabihf-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include -I/usr/local/include -I/opt/include -I/opt/local/include -I/usr/lib/python3/dist-packages/numpy/core/include -I/usr/include/python3.7m -c talib/_ta_lib.c -o build/temp.linux-armv7l-3.7/talib/_ta_lib.o
    talib/_ta_lib.c:611:10: fatal error: ta-lib/ta_defs.h: File o directory non esistente
     #include "ta-lib/ta_defs.h"
              ^~~~~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-fugao5k4/TA-Lib/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-_opvumy8/install-record.txt --single-version-externally-managed --compile --
  ```

**Comment by xmatthias** [comment]
you need to install the c library first. 
ta-lib consists of 2 parts, the C part, and the python part.

Installing the python part without the C part will fail. 

The [ta lib python docs](http://mrjbq7.github.io/ta-lib/install.html) have a decent instruction - alternatively you can use the quick install script (`./setup.sh -i`) to install the c part (this will take a while on a raspberry).

You can also use the [docker image](https://www.freqtrade.io/en/latest/docker_quickstart/) we provide to get started with freqtrade.

**Comment by xmatthias** [comment]
I'll assume this as answered by the above.

Feel free to comment below / reopen if that's not the case.

**Comment by ssoydan1973** [comment]
Use whl file according to yout python version. And install manually. If your python is 3.7 you need to download version 37.
https://www.lfd.uci.edu/~gohlke/pythonlibs/#ta-lib

**Comment by gitHubUser151** [comment]
this helped 👍  :
https://blog.quantinsti.com/install-ta-lib-python/#windows

**Comment by allenyllee** [comment]
try `conda install -c conda-forge ta-lib`

**Comment by clockwiser** [comment]
> try `conda install -c conda-forge ta-lib`

This works, unbelievable.

**Comment by HappyCthulhu** [comment]
> try `conda install -c conda-forge ta-lib`

this is the only solution, that helped me. I tried everything)
btw, i made it with `sudo`. Only with root priviliges it works on my Arch pc

**Comment by groundedtako** [comment]
> you need to install the c library first. ta-lib consists of 2 parts, the C part, and the python part.
> 
> Installing the python part without the C part will fail.
> 
> The [ta lib python docs](http://mrjbq7.github.io/ta-lib/install.html) have a decent instruction - alternatively you can use the quick install script (`./setup.sh -i`) to install the c part (this will take a while on a raspberry).
> 
> You can also use the [docker image](https://www.freqtrade.io/en/latest/docker_quickstart/) we provide to get started with freqtrade.

can confirm that this is a fix for mac.
TLDR: `brew install ta-lib.`
Note it takes a while

**Comment by Dre-AsiliVentures** [comment]
The solution that worked for me was to install another TA-Lib library with same functionalities (unwrapped version) through pip install TA-Lib-Precompiled

**Comment by sainipankaj15** [comment]
> this helped 👍 : https://blog.quantinsti.com/install-ta-lib-python/#windows

Thank you man !!.
I reached from where I started

Note : If you guys facing issue in Step 1: Install Ta-Lib dependencies (In Linux system)
so instead of using "wget http://prdownloads.sourceforge.net/ta-lib/ta-lib-0.4.0-src.tar.gz"
use "wget https://prdownloads.sourceforge.net/ta-lib/ta-lib-0.4.0-src.tar.gz"

It will work. You might be face this issue.
